### PR TITLE
Question with responses

### PR DESCRIPTION
**Pull request by athavy92**

### CODE REVIEW

File: src/app/models/Response.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
+import { Question } from '../../app/models/Question';
+
 export interface Response {
   id: number;
   responderId: number;
   questionId: number;
   body: string;
   creationDate: string;
+  question: Question[];
```
Comment (AdamDKing) [inline src/app/models/Response.ts:9]: why array of questions?

File: src/app/services/question.service.ts [hunk 2]
```diff
@@ -43,6 +47,12 @@ export class QuestionService {
     return this.http.put<Question>(url, question, httpOptions);
   }
 
+  highlightResponse(question: Question): Observable<Question> {
```
Comment (AdamDKing) [inline src/app/services/question.service.ts:49]: this is a generic update method.  Either change functionality to match the name or change the name to match the functionality

File: src/app/static/response/response.component.html [hunk 3]
```diff
@@ -1,10 +1,15 @@
 <mat-card class="response-card">
   <mat-card-header>
     <div mat-card-avatar class="example-header-image"></div>
-    <mat-card-title>{{ response.responderId }}</mat-card-title>
-    <mat-card-subtitle>{{ response.creationDate | date: 'short' }}</mat-card-subtitle>
+    <mat-card-title>Response ID: {{ response.id }}</mat-card-title>
+    <mat-card-subtitle>Question ID: {{ response.questionId }}</mat-card-subtitle>
+    <mat-card-subtitle>Responder ID: {{ response.responderId }}</mat-card-subtitle>
```
Comment (AdamDKing) [inline src/app/static/response/response.component.html:6]: why display these to the user?

File: src/app/static/highlighted-response/highlighted-response.component.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<mat-card class="highlight-response-card">
+  <mat-card-header>
+    <div mat-card-avatar class="example-header-image"></div>
+    <mat-card-title>Response ID: {{ highlight.id }}</mat-card-title>
+    <mat-card-subtitle>Question ID: {{ highlight.questionId }}</mat-card-subtitle>
+    <mat-card-subtitle>Responder ID: {{ highlight.responderId }}</mat-card-subtitle>
```
Comment (AdamDKing) [inline src/app/static/highlighted-response/highlighted-response.component.html:6]: why display these to the user?

File: src/app/static/response/response.component.ts [hunk 5]
```diff
@@ -1,21 +1,47 @@
...
+      .subscribe(
+        data => {
+          console.log('PATCH successful', data);
+          // Will add snackbar or other notification here
+        },
+        error => {
+          console.log('PATCH ERROR', error);
+        },
+      );
+  };
```
Comment (AdamDKing) [inline src/app/static/response/response.component.ts]: Work with the backend to get this resolved.  You're telling the API to update the question to all null fields, so either hit a different endpoint or send a version of the question that doesn't update all fields to null

File: src/app/static/view-question/view-question.component.ts [hunk 6]
```diff
@@ -17,25 +15,48 @@ export class ViewQuestionComponent implements OnInit {
...
   ngOnInit() {
+    const selectedQuestionId = this.questionService.getQuestionId();
+
     this.questionService.getQuestions().subscribe(questions => {
       for (const q in questions) {
-        if (this.questionService.getQuestionId() === questions[q].id) {
+        if (selectedQuestionId === questions[q].id) {
           this.selectedQuestion = questions[q];
         }
       }
```
Comment (AdamDKing) [inline src/app/static/view-question/view-question.component.ts]: why not just get the question by its id?  You have an endpoint for that

**Review by AdamDKing — CHANGES_REQUESTED** [review]

File: src/app/models/Response.ts (hunk 1 above)
Comment (athavy92) [reply src/app/models/Response.ts:9]: I just checked. It shouldn't be an array

File: src/app/static/response/response.component.ts (hunk 5 above)
Comment (athavy92) [reply src/app/static/response/response.component.ts]: Got an endpoint that works and the issue is resolved.

File: src/app/static/view-question/view-question.component.ts (hunk 6 above)
Comment (athavy92) [reply src/app/static/view-question/view-question.component.ts]: Refactored to get question by the ID of the selected question

**Pull request merged**